### PR TITLE
[RF-26747] Handle paginated responses from backend when fetching sites and environments

### PR DESCRIPTION
**Pull request by magni-**
Pagination code is copy-pasted from what we already do when fetching features and folders.

### CODE REVIEW

File: rainforest/resources.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (c *Client) getPaginatedResource(endpoint string, coll interface{}, collect
 
 	totalPagesHeader := res.Header.Get("X-Total-Pages")
 	if totalPagesHeader == "" {
-		return fmt.Errorf("Missing X-Total-Pages header in HTTP Response")
+		totalPagesHeader = "1"
```
Comment (ubergeek42) [inline rainforest/resources.go:39]: Do the newly paginated endpoints not have a total pages header? Or why this change? It seems fine, just curious.

Comment (jbarber) [inline rainforest/resources.go:39]: Is this because /sites + /environments are not paginated yet?

**Review by jbarber — APPROVED** [review]
Tests, or already has good coverage?

**Review by ubergeek42 — APPROVED** [review]
Looks ok to me

File: rainforest/resources.go (hunk 1 above)
Comment (magni-) [reply rainforest/resources.go:39]: Exactly, this is to make the CLI forward compatible before adding pagination to those endpoints.

Comment (magni-) [reply rainforest/resources.go:39]: See response to @jbarber above, they're not paginated _yet_. Once they are, they'll have pagination headers like all the other paginated endpoints.

**Comment by magni-** [comment]
> Tests, or already has good coverage?

There's already tests for `GetEnvironments` and `GetSites`, so they cover that switching to use `getPaginatedResource` doesn't break when the endpoints don't respond with an `X-Total-Pages` header. Actual pagination is tested in the resources that are already paginated (`GetFeatures, `GetFolders, `GetRunGroups`).
